coverage_utils.py

```diff
@@ -10,26 +11,54 @@
 from core.metadata_layer import ReplacementPolicy
 from core.mirror import MirrorUploader
 from core.model import (
+    get_one,
     DataSource,
+    ExternalIntegration,
+    ExternalIntegrationLink,
     Work,
 )
 
+class MetadataWranglerReplacementPolicy(ReplacementPolicy):
+    """A ReplacementPolicy that uses the only configured storage
+    integration as its cover mirror.
+    """
+    @classmethod
+    def from_db(cls, _db, **kwargs):
+        """Create a MetadataWranglerReplacementPolicy based on
+        database settings -- specifically, with the configured
+        MirrorUploader for cover images.
+
+        :param mirror: Pass in a mock MirrorUploader to use it
+        instead of creating a real one.
+        """
+        mirror = kwargs.pop('mirror', None)
+        if not mirror:
+            integration = get_one(
+                _db, ExternalIntegration, goal=ExternalIntegration.STORAGE_GOAL
+            )
+            if integration:
+                mirror = MirrorUploader.implementation(integration)
+            else:
+                logging.error(
+                    "No storage integration is configured. Cover images will not be mirrored. You really ought to set up a storage integration."
+                )
+                mirror = None
```

Comment:
In here we keep all the details about how exactly we create `MirrorUploader` objects for a metadata wrangler. Then we just need to make sure we always make a `MetadataWranglerReplacementPolicy` instead of a normal `ReplacementPolicy`. If anything else like this shows up in the future we can put it in this class as well.